### PR TITLE
Added and reconfigured globalization tests

### PR DESCRIPTION
**Pull request by sostanko**

### CODE REVIEW

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1304SpecifyCultureInfoTests.cs [hunk 1]
```diff
@@ -33,10 +33,35 @@ void M(string strA, string strB)
...
+                                                 preferred: "Public Overloads Function ToUpper(culture As System.Globalization.CultureInfo) As String"));
+
+        }
+
         [Fact]
-        public void CA1304ShoudUseOverloadsWithExplicitCultureInfoParameterTests_Misc_CS()
+        public void CA1304CultureInfoParameterTests_Misc()
         {
             VerifyCSharp(@"
 using System;
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1304SpecifyCultureInfoTests.cs]: Are you missing something here? Only one using statement?

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1304SpecifyCultureInfoTests.cs [hunk 2]
```diff
@@ -229,12 +222,13 @@ End Sub
         Console.WriteLine(String.Format(provider, format, content))
     End Sub
 End Module",
-            GetCA1304BasicDefaultResultAt(10, 9, callee: "Friend Sub M1(format As String, content As String)", 
-                                                 caller: "Public Sub M(f As String, s As String)", 
+            GetCA1304BasicDefaultResultAt(10, 9, callee: "Friend Sub M1(format As String, content As String)",
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1304SpecifyCultureInfoTests.cs]: Report diagnostics on Line 10? There is no line 10.

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1304SpecifyCultureInfoTests.cs [hunk 3]
```diff
@@ -229,12 +222,13 @@ End Sub
         Console.WriteLine(String.Format(provider, format, content))
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1304SpecifyCultureInfoTests.cs]: Are you missing a Public Sub definition or something?

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs [hunk 4]
```diff
@@ -22,28 +20,54 @@ sealed class C
     void M(string strA, string strB)
     {
         string str1 = string.Format(""Foo {0}"", strA);
-        string str2 = string.Format(""Foo {0} {1}"", strA, strB);
+        string str2 = string.Format(""Foo {0} {1}"", strA, strB); 
+    }
+}",
+            GetCA1305CSharpDefaultResultAt(9, 23, MessageAlternateString,
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs]: Line numbers looks incorrect here and right below

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs [hunk 5]
```diff
@@ -68,24 +92,111 @@ private static void LeadingIFormatProvider2(IFormatProvider provider, string s)
             Console.WriteLine(string.Format(provider, s));
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs]: Missing a few lines here. The class definition, and a method definition

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs [hunk 6]
```diff
@@ -150,20 +299,6 @@ void M(string strA)
     {
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs]: Missing class and function definition

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs [hunk 7]
```diff
@@ -173,53 +308,201 @@ void M(string strA)
...
+            GetCA1305BasicDefaultResultAt(10, 30, MessageUICultureString,
+                                                  "Public Sub M(strA As String, strB As String)",
+                                                  "CultureInfo.InstalledUICulture",
+                                                  "Public Shared Overloads Function Format(provider As System.IFormatProvider, format As String, arg0 As Object, arg1 As Object) As String"));
+        }
+
+        [Fact]
+        public void CA1305IFormatProvider_LeadingCultureInfo()
+        {
+            VerifyCSharp(@"using System;
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs]: Start to see some inconsistent style here. Shall we put the first statement on the next line of '@"' or same line? Once we agrees, we can follow this convention.

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1307SpecifyStringComparisonTests.cs [hunk 8]
```diff
@@ -25,15 +24,7 @@ void M(string strA, string strB)
         int r; // 10
         r =  String.Compare(strA, strB);
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1307SpecifyStringComparisonTests.cs]: Missing class definition and method def.

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1309UseOrdinalStringComparisonTests.cs [hunk 9]
```diff
@@ -6,14 +6,15 @@
 using Xunit;
 
 namespace System.Globalization.Analyzers.UnitTests
-{                              
+{
     public sealed class CA1309UseOrdinalStringComparisonTests : DiagnosticAnalyzerTestBase
     {
 #pragma warning disable CS0219
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1309UseOrdinalStringComparisonTests.cs]: Can I ask why disable it? I can't find any un-assigned variables.

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs (hunk 4 above)
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs]: That's weird, These tests shouldn't be working. Did these test cases pass when you run them?

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1307SpecifyStringComparisonTests.cs [hunk 10]
```diff
@@ -54,16 +45,16 @@ void M(string strA, string strB)
...
-            GetCA1307CSharpDefaultResultAt(12, 14, "", "", ""),
-            GetCA1307CSharpDefaultResultAt(13, 14, "", "", ""),
-            GetCA1307CSharpDefaultResultAt(14, 14, "", "", ""),
-            GetCA1307CSharpDefaultResultAt(15, 14, "", "", ""),
-            GetCA1307CSharpDefaultResultAt(16, 14, "", "", ""),
-            GetCA1307CSharpDefaultResultAt(17, 14, "", "", ""),
-            GetCA1307CSharpDefaultResultAt(18, 14, "", "", ""),
-            GetCA1307CSharpDefaultResultAt(19, 18, "", "", ""),
-            GetCA1307CSharpDefaultResultAt(20, 18, "", "", ""));
+            GetCA1307CSharpDefaultResultAt(11, 14,
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1307SpecifyStringComparisonTests.cs]: After I counted the rows, I didn't find row 11. Did this ever test pass?

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1307SpecifyStringComparisonTests.cs [hunk 11]
```diff
@@ -106,16 +89,231 @@ Imports System.Globalization
...
-            GetCA1307BasicDefaultResultAt(12, 14, "", "", ""),
-            GetCA1307BasicDefaultResultAt(13, 14, "", "", ""),
-            GetCA1307BasicDefaultResultAt(14, 14, "", "", ""),
-            GetCA1307BasicDefaultResultAt(15, 14, "", "", ""),
-            GetCA1307BasicDefaultResultAt(16, 14, "", "", ""),
-            GetCA1307BasicDefaultResultAt(17, 14, "", "", ""),
-            GetCA1307BasicDefaultResultAt(18, 14, "", "", ""),
-            GetCA1307BasicDefaultResultAt(19, 18, "", "", ""),
-            GetCA1307BasicDefaultResultAt(20, 18, "", "", ""));
+            GetCA1307BasicDefaultResultAt(11, 14,
```
Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1307SpecifyStringComparisonTests.cs]: Line number here too looks impossible. Can you check all the similar issues?

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1304SpecifyCultureInfoTests.cs (hunk 1 above)
Comment (sostanko) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1304SpecifyCultureInfoTests.cs:67]: Make sure you expand the code during review. Line 67 is "using System;". Lines 68-71 are other using statements.

Comment (qinxgit) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1304SpecifyCultureInfoTests.cs:67]: Got it.

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs (hunk 7 above)
Comment (sostanko) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs]: In most of cases I see that after @" there is an empty line, and then the code starts. Let's stick with that as a convention.

File: src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1309UseOrdinalStringComparisonTests.cs (hunk 9 above)
Comment (sostanko) [inline src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1309UseOrdinalStringComparisonTests.cs]: Not sure. It was there in original template .cs file, so I just kept it.
It's removed now.

**Pull request merged**